mono/utils/mono-threads-coop.c

```diff
@@ -99,6 +99,12 @@ copy_stack_data (MonoThreadInfo *info, void* stackdata_begin)
 	state = &info->thread_saved_state [SELF_SUSPEND_STATE_INDEX];
 
 	stackdata_size = (char*)stackdata_begin - (char*)stackdata_end;
+
+	if (((guint64) stackdata_begin & (SIZEOF_VOID_P - 1)) != 0)
```

Comment:
you should use gsize here, it's always going to be the size of a pointer this way